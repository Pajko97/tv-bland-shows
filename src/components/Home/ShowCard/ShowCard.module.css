.show_card {
  font-family: "Red Hat Display";
  -webkit-box-shadow: 5px 5px 15px 5px rgba(0, 0, 0, 0.46);
  box-shadow: 5px 5px 15px 5px rgba(0, 0, 0, 0.46);
  max-width: 13rem;
}

.show_card p {
  font-family: "Poppins";
  padding: 1rem;
  font-weight: 500;
  font-size: 1.2rem;
}
.show_card img {
  width: auto;
  max-width: 13rem;
}

.rating {
  display: flex;
  align-items: center;
  justify-content: flex-start;
  padding: 1rem;
}

@media only screen and (max-width: 800px) {
  .show_card {
    max-width: 13rem;
  }
  .rating {
    flex-direction: column;
  }
  .show_card img {
    max-width: 13rem;
  }

}
