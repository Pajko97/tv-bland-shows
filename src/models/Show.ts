interface SingleShow {
    averageRuntime: number,
    dvdCountry: null
    ended: string,
    externals: Object
    genres: Array<any>,
    id: number,
    image: {
        medium : string,
        original : string
    },
    language: string
    name: string,
    network: {
        country : object,
        id: number,
        name : string
    },
    officialSite: null,
    premiered: string,
    rating: {
        average : any
    },
    runtime: number,
    schedule: {
        time : string,
        days: [string]
    },
    status: String,
    summary: string,
    type: string,
    updated: number,
    url: string,
    webChannel: null,
    weight: number,
    _links: Object
}   

export default SingleShow