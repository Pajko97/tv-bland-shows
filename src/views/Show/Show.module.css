.show_wrapper {
  padding: 10rem;
  font-family: "Red Hat Display";
}

.show_header {
  display: flex;
}
.show_header img {
  margin-right: 5rem;
  min-height: 100%;
  max-width: 25rem;
  max-height: auto;
}

.show_basic_info {
  max-width: 40rem;
}
.show_basic_info p {
  line-height: 150%;
}

.show_basic_info .rate {
  margin-left: 0.5rem;
}

.show_details {
  display: flex;
  justify-content: space-between;
  align-items: flex-start;
}

@media only screen and (max-width: 800px) {
  .show_wrapper {
    padding: 3rem;
  }
  .show_basic_info {
    margin-top: 2rem;
  }
  .show_header {
    display: flex;
    flex-direction: column;
  }
  .show_header img {
    margin-right: 0;
  }
  .show_details {
    flex-direction: column;
  }
}
