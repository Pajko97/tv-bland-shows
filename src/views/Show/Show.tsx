import React, { useEffect, useState } from 'react';
import { useParams } from 'react-router';
import API from '../../api/shows-api';
import SingleShow from '../../models/Show'
import striptags from '../../helpers/striptags'
import Rating from 'react-rating'
import ActorList from "../../components/Show/ActorList/ActorList"
import style from './Show.module.css'
import ShowInfo from '../../components/Show/ShowInfo/ShowInfo';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { faCoffee, faStar, faStarHalfStroke } from '@fortawesome/free-solid-svg-icons'

type Props = Object

function Show(props: Props) {
  const [show, setShow] = useState<SingleShow>()

  let { id }: any = useParams();
  let showId = parseInt(id)

  useEffect(() => {
    let api = new API();
    api.getShow(showId)
      .then(res => setShow(res.data))
  }, [])

  const calcRating = (rating: number) =>
	rating > 5 ? Math.round((rating / 2) * 10) / 10 : rating

  return (
    <React.Fragment>
      {show && <div className={style.show_wrapper}>
        <div>
          <div className={style.show_header}>
            <img src={show.image.medium} alt="show cover"></img>
            <div className={style.show_basic_info}>
            <Rating
              emptySymbol={<FontAwesomeIcon icon={faStarHalfStroke} />}
              readonly
              initialRating={calcRating(show.rating.average)}
              fullSymbol={<FontAwesomeIcon icon={faStar}/>}

            />
            <span className={style.rate}>{show.rating.average ? `${show.rating.average}/5` : 'Not provided'}</span>
            <h1>{show.name}</h1>
            <p>{striptags(show.summary)}</p>
            </div>
          </div>

          <div className={style.show_details}>
            <ShowInfo show={show} />
            <ActorList showId={showId} />
          </div>
        </div>
      </div>}
    </React.Fragment>
  );
}

export default Show

