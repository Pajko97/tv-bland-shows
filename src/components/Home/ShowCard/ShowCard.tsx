import Rating from "react-rating";
import { Link } from "react-router-dom";
import Show from "../../../models/Show";
import style from "./ShowCard.module.css";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faStar, faStarHalfAlt } from "@fortawesome/free-solid-svg-icons";
type Props = {
  data: Show;
};

const calcRating = (rating: number) =>
  rating > 5 ? Math.round((rating / 2) * 10) / 10 : rating;

export default function ShowCard(props: Props) {
  return (
    <div className={style.show_card}>
      <Link to={`/shows/${props.data.id}`}>
        <img
          src={
            props.data.image
              ? props.data.image.medium
              : "https://via.placeholder.com/250x350"
          }
          alt={props.data.name}
        />
      </Link>
      <div className={style.rating}>
        <Rating
          emptySymbol={<FontAwesomeIcon icon={faStarHalfAlt} />}
          readonly
          initialRating={calcRating(props.data.rating.average)}
          fullSymbol={<FontAwesomeIcon icon={faStar} />}
        />
      </div>
      <p>{props.data.name}</p>
    </div>
  );
}
