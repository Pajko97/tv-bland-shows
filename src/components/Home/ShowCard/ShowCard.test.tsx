import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import API from '../../../api/shows-api';
import Show from '../../../models/Show';
import ShowCard from './ShowCard';

it('render app', async () => {
  let render, show: Show;
  let api = new API()
  await api.getShow(2)
    .then((res) => {
      show = res.data;
      render = renderer.create(
        <BrowserRouter>
          <ShowCard data={show} />
        </BrowserRouter>
        ).toJSON();
    });
    
  expect(render).toMatchSnapshot();
})