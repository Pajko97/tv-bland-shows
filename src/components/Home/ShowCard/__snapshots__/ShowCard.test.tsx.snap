// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render app 1`] = `
<div
  className="show_card"
>
  <a
    href="/shows/2"
    onClick={[Function]}
  >
    <img
      alt="Person of Interest"
      src="https://static.tvmaze.com/uploads/images/medium_portrait/163/407679.jpg"
    />
  </a>
  <div
    className="rating"
  >
    <span
      style={
        Object {
          "direction": "ltr",
          "display": "inline-block",
        }
      }
    >
      <span
        onClick={[Function]}
        onMouseMove={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        style={
          Object {
            "cursor": "inherit",
            "display": "inline-block",
            "position": "relative",
          }
        }
      >
        <span
          style={
            Object {
              "visibility": "hidden",
            }
          }
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-star-half-stroke "
            data-icon="star-half-stroke"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 576 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M463.1 474.7C465.1 486.7 460.2 498.9 450.2 506C440.3 513.1 427.2 514 416.5 508.3L288.1 439.8L159.8 508.3C149 514 135.9 513.1 126 506C116.1 498.9 111.1 486.7 113.2 474.7L137.8 328.1L33.58 225.9C24.97 217.3 21.91 204.7 25.69 193.1C29.46 181.6 39.43 173.2 51.42 171.5L195 150.3L259.4 17.97C264.7 6.954 275.9-.0391 288.1-.0391C300.4-.0391 311.6 6.954 316.9 17.97L381.2 150.3L524.9 171.5C536.8 173.2 546.8 181.6 550.6 193.1C554.4 204.7 551.3 217.3 542.7 225.9L438.5 328.1L463.1 474.7zM288 376.4L288.1 376.3L399.7 435.9L378.4 309.6L469.2 219.8L343.8 201.4L288.1 86.85L288 87.14V376.4z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </span>
        <span
          style={
            Object {
              "display": "inline-block",
              "left": 0,
              "overflow": "hidden",
              "position": "absolute",
              "top": 0,
              "width": "100%",
            }
          }
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-star "
            data-icon="star"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 576 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M381.2 150.3L524.9 171.5C536.8 173.2 546.8 181.6 550.6 193.1C554.4 204.7 551.3 217.3 542.7 225.9L438.5 328.1L463.1 474.7C465.1 486.7 460.2 498.9 450.2 506C440.3 513.1 427.2 514 416.5 508.3L288.1 439.8L159.8 508.3C149 514 135.9 513.1 126 506C116.1 498.9 111.1 486.7 113.2 474.7L137.8 328.1L33.58 225.9C24.97 217.3 21.91 204.7 25.69 193.1C29.46 181.6 39.43 173.2 51.42 171.5L195 150.3L259.4 17.97C264.7 6.954 275.9-.0391 288.1-.0391C300.4-.0391 311.6 6.954 316.9 17.97L381.2 150.3z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </span>
      </span>
      <span
        onClick={[Function]}
        onMouseMove={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        style={
          Object {
            "cursor": "inherit",
            "display": "inline-block",
            "position": "relative",
          }
        }
      >
        <span
          style={
            Object {
              "visibility": "hidden",
            }
          }
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-star-half-stroke "
            data-icon="star-half-stroke"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 576 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M463.1 474.7C465.1 486.7 460.2 498.9 450.2 506C440.3 513.1 427.2 514 416.5 508.3L288.1 439.8L159.8 508.3C149 514 135.9 513.1 126 506C116.1 498.9 111.1 486.7 113.2 474.7L137.8 328.1L33.58 225.9C24.97 217.3 21.91 204.7 25.69 193.1C29.46 181.6 39.43 173.2 51.42 171.5L195 150.3L259.4 17.97C264.7 6.954 275.9-.0391 288.1-.0391C300.4-.0391 311.6 6.954 316.9 17.97L381.2 150.3L524.9 171.5C536.8 173.2 546.8 181.6 550.6 193.1C554.4 204.7 551.3 217.3 542.7 225.9L438.5 328.1L463.1 474.7zM288 376.4L288.1 376.3L399.7 435.9L378.4 309.6L469.2 219.8L343.8 201.4L288.1 86.85L288 87.14V376.4z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </span>
        <span
          style={
            Object {
              "display": "inline-block",
              "left": 0,
              "overflow": "hidden",
              "position": "absolute",
              "top": 0,
              "width": "100%",
            }
          }
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-star "
            data-icon="star"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 576 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M381.2 150.3L524.9 171.5C536.8 173.2 546.8 181.6 550.6 193.1C554.4 204.7 551.3 217.3 542.7 225.9L438.5 328.1L463.1 474.7C465.1 486.7 460.2 498.9 450.2 506C440.3 513.1 427.2 514 416.5 508.3L288.1 439.8L159.8 508.3C149 514 135.9 513.1 126 506C116.1 498.9 111.1 486.7 113.2 474.7L137.8 328.1L33.58 225.9C24.97 217.3 21.91 204.7 25.69 193.1C29.46 181.6 39.43 173.2 51.42 171.5L195 150.3L259.4 17.97C264.7 6.954 275.9-.0391 288.1-.0391C300.4-.0391 311.6 6.954 316.9 17.97L381.2 150.3z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </span>
      </span>
      <span
        onClick={[Function]}
        onMouseMove={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        style={
          Object {
            "cursor": "inherit",
            "display": "inline-block",
            "position": "relative",
          }
        }
      >
        <span
          style={
            Object {
              "visibility": "hidden",
            }
          }
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-star-half-stroke "
            data-icon="star-half-stroke"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 576 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M463.1 474.7C465.1 486.7 460.2 498.9 450.2 506C440.3 513.1 427.2 514 416.5 508.3L288.1 439.8L159.8 508.3C149 514 135.9 513.1 126 506C116.1 498.9 111.1 486.7 113.2 474.7L137.8 328.1L33.58 225.9C24.97 217.3 21.91 204.7 25.69 193.1C29.46 181.6 39.43 173.2 51.42 171.5L195 150.3L259.4 17.97C264.7 6.954 275.9-.0391 288.1-.0391C300.4-.0391 311.6 6.954 316.9 17.97L381.2 150.3L524.9 171.5C536.8 173.2 546.8 181.6 550.6 193.1C554.4 204.7 551.3 217.3 542.7 225.9L438.5 328.1L463.1 474.7zM288 376.4L288.1 376.3L399.7 435.9L378.4 309.6L469.2 219.8L343.8 201.4L288.1 86.85L288 87.14V376.4z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </span>
        <span
          style={
            Object {
              "display": "inline-block",
              "left": 0,
              "overflow": "hidden",
              "position": "absolute",
              "top": 0,
              "width": "100%",
            }
          }
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-star "
            data-icon="star"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 576 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M381.2 150.3L524.9 171.5C536.8 173.2 546.8 181.6 550.6 193.1C554.4 204.7 551.3 217.3 542.7 225.9L438.5 328.1L463.1 474.7C465.1 486.7 460.2 498.9 450.2 506C440.3 513.1 427.2 514 416.5 508.3L288.1 439.8L159.8 508.3C149 514 135.9 513.1 126 506C116.1 498.9 111.1 486.7 113.2 474.7L137.8 328.1L33.58 225.9C24.97 217.3 21.91 204.7 25.69 193.1C29.46 181.6 39.43 173.2 51.42 171.5L195 150.3L259.4 17.97C264.7 6.954 275.9-.0391 288.1-.0391C300.4-.0391 311.6 6.954 316.9 17.97L381.2 150.3z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </span>
      </span>
      <span
        onClick={[Function]}
        onMouseMove={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        style={
          Object {
            "cursor": "inherit",
            "display": "inline-block",
            "position": "relative",
          }
        }
      >
        <span
          style={
            Object {
              "visibility": "hidden",
            }
          }
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-star-half-stroke "
            data-icon="star-half-stroke"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 576 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M463.1 474.7C465.1 486.7 460.2 498.9 450.2 506C440.3 513.1 427.2 514 416.5 508.3L288.1 439.8L159.8 508.3C149 514 135.9 513.1 126 506C116.1 498.9 111.1 486.7 113.2 474.7L137.8 328.1L33.58 225.9C24.97 217.3 21.91 204.7 25.69 193.1C29.46 181.6 39.43 173.2 51.42 171.5L195 150.3L259.4 17.97C264.7 6.954 275.9-.0391 288.1-.0391C300.4-.0391 311.6 6.954 316.9 17.97L381.2 150.3L524.9 171.5C536.8 173.2 546.8 181.6 550.6 193.1C554.4 204.7 551.3 217.3 542.7 225.9L438.5 328.1L463.1 474.7zM288 376.4L288.1 376.3L399.7 435.9L378.4 309.6L469.2 219.8L343.8 201.4L288.1 86.85L288 87.14V376.4z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </span>
        <span
          style={
            Object {
              "display": "inline-block",
              "left": 0,
              "overflow": "hidden",
              "position": "absolute",
              "top": 0,
              "width": "100%",
            }
          }
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-star "
            data-icon="star"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 576 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M381.2 150.3L524.9 171.5C536.8 173.2 546.8 181.6 550.6 193.1C554.4 204.7 551.3 217.3 542.7 225.9L438.5 328.1L463.1 474.7C465.1 486.7 460.2 498.9 450.2 506C440.3 513.1 427.2 514 416.5 508.3L288.1 439.8L159.8 508.3C149 514 135.9 513.1 126 506C116.1 498.9 111.1 486.7 113.2 474.7L137.8 328.1L33.58 225.9C24.97 217.3 21.91 204.7 25.69 193.1C29.46 181.6 39.43 173.2 51.42 171.5L195 150.3L259.4 17.97C264.7 6.954 275.9-.0391 288.1-.0391C300.4-.0391 311.6 6.954 316.9 17.97L381.2 150.3z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </span>
      </span>
      <span
        onClick={[Function]}
        onMouseMove={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        style={
          Object {
            "cursor": "inherit",
            "display": "inline-block",
            "position": "relative",
          }
        }
      >
        <span
          style={Object {}}
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-star-half-stroke "
            data-icon="star-half-stroke"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 576 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M463.1 474.7C465.1 486.7 460.2 498.9 450.2 506C440.3 513.1 427.2 514 416.5 508.3L288.1 439.8L159.8 508.3C149 514 135.9 513.1 126 506C116.1 498.9 111.1 486.7 113.2 474.7L137.8 328.1L33.58 225.9C24.97 217.3 21.91 204.7 25.69 193.1C29.46 181.6 39.43 173.2 51.42 171.5L195 150.3L259.4 17.97C264.7 6.954 275.9-.0391 288.1-.0391C300.4-.0391 311.6 6.954 316.9 17.97L381.2 150.3L524.9 171.5C536.8 173.2 546.8 181.6 550.6 193.1C554.4 204.7 551.3 217.3 542.7 225.9L438.5 328.1L463.1 474.7zM288 376.4L288.1 376.3L399.7 435.9L378.4 309.6L469.2 219.8L343.8 201.4L288.1 86.85L288 87.14V376.4z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </span>
        <span
          style={
            Object {
              "display": "inline-block",
              "left": 0,
              "overflow": "hidden",
              "position": "absolute",
              "top": 0,
              "width": "50%",
            }
          }
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-star "
            data-icon="star"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 576 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M381.2 150.3L524.9 171.5C536.8 173.2 546.8 181.6 550.6 193.1C554.4 204.7 551.3 217.3 542.7 225.9L438.5 328.1L463.1 474.7C465.1 486.7 460.2 498.9 450.2 506C440.3 513.1 427.2 514 416.5 508.3L288.1 439.8L159.8 508.3C149 514 135.9 513.1 126 506C116.1 498.9 111.1 486.7 113.2 474.7L137.8 328.1L33.58 225.9C24.97 217.3 21.91 204.7 25.69 193.1C29.46 181.6 39.43 173.2 51.42 171.5L195 150.3L259.4 17.97C264.7 6.954 275.9-.0391 288.1-.0391C300.4-.0391 311.6 6.954 316.9 17.97L381.2 150.3z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </span>
      </span>
    </span>
  </div>
  <p>
    Person of Interest
  </p>
</div>
`;
